src/pubsub/message/sign.js

```diff
@@ -0,0 +1,92 @@
+'use strict'
+
+const PeerId = require('peer-id')
+const { Message } = require('./index')
+const uint8ArrayConcat = require('uint8arrays/concat')
+const uint8ArrayFromString = require('uint8arrays/from-string')
+const SignPrefix = uint8ArrayFromString('libp2p-pubsub:')
+
+/**
+ * Signs the provided message with the given `peerId`
+ *
+ * @param {PeerId} peerId
+ * @param {Message} message
+ * @returns {Promise<Message>}
+ */
+async function signMessage (peerId, message) {
+  // Get the message in bytes, and prepend with the pubsub prefix
+  const bytes = uint8ArrayConcat([
+    SignPrefix,
+    Message.encode(message)
+  ])
+
+  const signature = await peerId.privKey.sign(bytes)
+
+  return {
+    ...message,
+    signature: signature,
+    key: peerId.pubKey.bytes
+  }
+}
+
+/**
+ * Verifies the signature of the given message
+ * @param {InMessage} message
+ * @returns {Promise<Boolean>}
+ */
+async function verifySignature (message) {
+  // Get message sans the signature
+  const baseMessage = { ...message }
+  delete baseMessage.signature
+  delete baseMessage.key
+  if (typeof baseMessage.from === 'string') {
```

Comment:
Won't this always be a string as this takes an `InMessage`? The message is run through the normalizeIn function by this point. If this is not a string there's an issue.